### PR TITLE
AP_RangeFinder_PWM: correct integer division issue

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: libraries/AP_RangeFinder/AP_RangeFinder_PWM.cpp [hunk 1]
```diff
@@ -67,7 +67,7 @@ bool AP_RangeFinder_PWM::get_reading(uint16_t &reading_cm)
     if (value_us == 0 || sample_count == 0) {
         return false;
     }
-    reading_cm = (value_us/sample_count) * 1e-1f; // correct for LidarLite.  Parameter needed?
+    reading_cm = value_us/(sample_count * 10.0f); // correct for LidarLite.  Parameter needed?  Converts from m -> cm here
```
Comment (WickedShell) [inline libraries/AP_RangeFinder/AP_RangeFinder_PWM.cpp]: No `.0f`!

Comment (rmackay9) [reply libraries/AP_RangeFinder/AP_RangeFinder_PWM.cpp]: I actually like the 10.0f.

Comment (WickedShell) [reply libraries/AP_RangeFinder/AP_RangeFinder_PWM.cpp]: The problem is this is actually pure integer math into a return of integer, promoting to float in the middle and then implicit cast to int is wasting time. The whole bug here is that this expression should simply never be a float.

Comment (peterbarker) [reply libraries/AP_RangeFinder/AP_RangeFinder_PWM.cpp]: /me checks type of reading_cm

Whoops.

Fixed.

File: libraries/AP_RangeFinder/AP_RangeFinder_PWM.cpp [hunk 2]
```diff
@@ -67,7 +67,7 @@ bool AP_RangeFinder_PWM::get_reading(uint16_t &reading_cm)
     if (value_us == 0 || sample_count == 0) {
         return false;
     }
-    reading_cm = (value_us/sample_count) * 1e-1f; // correct for LidarLite.  Parameter needed?
+    reading_cm = value_us/(sample_count * 10); // correct for LidarLite.  Parameter needed?  Converts from m -> cm here
```
Comment (WickedShell) [inline libraries/AP_RangeFinder/AP_RangeFinder_PWM.cpp]: Alright, I know I'm one of those Americans who use weird units. But uh, dividing by 10 is not converting meters to centimeters, which makes the new comment realllllly confusing?

**Pull request merged**